### DISCUSSION TITLE
`Session.scalar_one()`

### DISCUSSION THREAD

**Discussion by ljiatu** [sqlalchemy/sqlalchemy#7100] [Ideas]
Now that we have `Session.scalar()` and `Session.scalars()`, would it make sense to add a `Session.scalar_one()` to mirror `AsyncResult.scalar_one()`?

**Comment by ljiatu** [comment]
Hi! Want to follow up to see if you guys think this is a good idea. Happy to send in a PR if so.

**Comment by CaselIT** [comment]
Hi,

Thanks for the suggestion. I was not particularly convinced by it, and also Mike is not really on board, because we want to return result objects, not ORM objects from the session.
The outlier is `.scalar()` that was already there, but that's more about selecting numeric values and settings and is more Connection oriented.

So long term we cant duplicate all of Result on Session, `scalars()` made sense since the return is still a `Result` and selecting a single orm table is a common use pattern.

In any case thanks for the suggestions!
